vessel-runtime/src/main/java/com/textnow/android/vessel/VesselImpl.kt

```diff
@@ -251,6 +251,10 @@ class VesselImpl(
     override fun <T : Any> typeNameOf(value: T): String {
         val name = value.javaClass.kotlin.qualifiedName
 
+        if (value is Function<*>) {
+            throw AssertionError("functions are not allowed.")
```

Comment:
Before the kotlin update I made in this PR there was a unit test to prevent this - however after the update it became necessary to specify functions vs anonymous classes here.